src/xenia/kernel/xam/xam_net.cc

```diff
@@ -1286,47 +1306,228 @@ dword_result_t XampXAuthStartup_entry(pointer_t<XAUTH_SETTINGS> setttings) {
 }
 DECLARE_XAM_EXPORT1(XampXAuthStartup, kNetworking, kStub);
 
+dword_result_t XampXAuthShutdown_entry(pointer_t<XAUTH_SETTINGS> setttings) {
+  return X_ERROR_SUCCESS;
+}
+DECLARE_XAM_EXPORT1(XampXAuthShutdown, kNetworking, kStub);
+
 dword_result_t NetDll_XHttpStartup_entry(dword_t caller, dword_t reserved,
                                          dword_t reserved_ptr) {
   return 1;
 }
 DECLARE_XAM_EXPORT1(NetDll_XHttpStartup, kNetworking, kStub);
 
-dword_result_t NetDll_XHttpDoWork_entry(dword_t caller, dword_t handle,
-                                        dword_t unk) {
-  XThread::SetLastError(0);
-  return 0;
+dword_result_t XamGetServiceEndpoint_entry(lpstring_t service_name,
+                                           lpstring_t service_endpoint,
+                                           dword_t service_endpoint_len,
+                                           dword_t overlapped) {
+  strncpy(service_endpoint, cvars::api_address.c_str(), service_endpoint_len);
```

Comment:
Could you explain what this "service_endpoint" is used for? It being set to the API address suggests to me that changes to the API are also needed.